html/pfappserver/lib/pfappserver/PacketFence/Controller/Configurator.pm

```diff
@@ -292,37 +292,45 @@ sub configuration :Chained('object') :PathPart('configuration') :Args(0) {
+        my (%config, $status, $general_ref, $alerting_ref, $advanced_ref);
         ($status, $general_ref) = $pf_model->read('general');
-        if(is_success($status)) {
+        if (is_success($status)) {
             ($status, $alerting_ref) = $pf_model->read('alerting');
             if (is_success($status)) {
-                @config{qw(
-                    general.domain
-                    general.hostname
-                    general.dhcpservers
-                    alerting.emailaddr
-                )} = (
-                    @$general_ref{qw(
-                        domain
-                        hostname
-                        dhcpservers
-                    )},
-                    @$alerting_ref{emailaddr}
-                );
-                $c->stash->{'config'} = \%config;
+                ($status, $advanced_ref) = $pf_model->read('advanced');
+                if (is_success($status)) {
+                    @config{qw(
+                                  general.domain
+                                  general.hostname
+                                  general.dhcpservers
+                                  alerting.emailaddr
+                                  advanced.hash_passwords
+                             )} = (
+                                   @$general_ref{qw(
+                                                       domain
+                                                       hostname
+                                                       dhcpservers
+                                                  )},
+                                   @$alerting_ref{emailaddr},
+                                   @$advanced_ref{hash_passwords}
+                                  );
+                    $c->stash->{'config'} = \%config;
+                }
             }
         }
     }
     elsif ($c->request->method eq 'POST') {
         # Save configuration
         my ( $status, $message ) = (HTTP_OK);
-        my $general_domain      = $c->request->params->{'general.domain'};
-        my $general_hostname    = $c->request->params->{'general.hostname'};
-        my $general_dhcpservers = $c->request->params->{'general.dhcpservers'};
-        my $alerting_emailaddr  = $c->request->params->{'alerting.emailaddr'};
-
-        unless ($general_domain && $general_hostname && $general_dhcpservers && $alerting_emailaddr) {
+        my $general_domain          = $c->request->params->{'general.domain'};
+        my $general_hostname        = $c->request->params->{'general.hostname'};
+        my $general_dhcpservers     = $c->request->params->{'general.dhcpservers'};
+        my $alerting_emailaddr      = $c->request->params->{'alerting.emailaddr'};
+        my $advanced_hash_passwords = $c->request->params->{'advanced.hash_passwords'};
+
+        unless ($general_domain && $general_hostname && $general_dhcpservers &&
+                $alerting_emailaddr &&
+                $advanced_hash_passwords) {
```

Comment:
The idea was to group the parameters by section (*general*, *alerting*, and *advanced*).